snapshot-labs/sx.js
# feat: update starknet-tx client to support Vanilla authenticator

## Summary

Closes #42
Closes #40

I used `0x5ad4bd1ca422953ac845ac7915ddfa93fe3fecec0ed8c86db5e294b0e18c9bd` instead of suggested `0x03a4cb1c6e4439e2ce2c43b565e6347f2f334cd3a387e86fb14f7655c9b6704a` as the latter has outdated ABI (missing `metadata_uri_string_len` in `propose` call).

Currently it uses Vanilla authenticator, strategy and executor.

## Test plan

- Run tests
- Verify against https://goerli.voyager.online/contract/0x5ad4bd1ca422953ac845ac7915ddfa93fe3fecec0ed8c86db5e294b0e18c9bd#readContract that your proposal and vote (for proposal 0x3) was registered.
